### PR TITLE
[Dashboard] Overview page

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR updates contract-related components to use `useContract` hook and `ThirdwebContract` type. 

### Detailed summary
- Updated components to use `useContract` hook and `ThirdwebContract` type
- Replaced `SmartContract` with `ThirdwebContract` in multiple files
- Improved contract data fetching and usage efficiency

> The following files were skipped due to too many changes: `apps/dashboard/src/contract-ui/tabs/overview/components/ContractChecklist.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4222?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4222** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4222?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4219** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4219?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4218** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4218?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4217** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4217?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4215** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4215?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4208** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4208?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4207** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4207?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4206** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4206?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4205** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4205?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4204** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4204?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4203** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4203?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4202** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4202?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4201** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4201?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/contract-ui/tabs/overview/components/ContractChecklist.tsx [hunk 1]
```diff
@@ -34,24 +32,19 @@ type Step = {
...
-  const batchesToReveal = useBatchesToReveal(contract);
-  const accounts = useAccounts(contract);
-  const sharedMetadata = useSharedMetadata(contract);
+  const nfts = useNFTs(contractQuery.contract, { count: 1 });
+  const erc721Claimed = useClaimedNFTSupply(contractQuery.contract);
+  const claimConditions = useClaimConditions(contractQuery.contract);
+  const erc20Supply = useTokenSupply(contractQuery.contract);
+  const batchesToReveal = useBatchesToReveal(contractQuery.contract);
+  const accounts = useAccounts(contractQuery.contract);
+  const sharedMetadata = useSharedMetadata(contractQuery.contract);
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/overview/components/ContractChecklist.tsx:47]: if ur doing this lets also move these to v5 then in this PR

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/contract-ui/tabs/overview/components/ContractChecklist.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/overview/components/ContractChecklist.tsx:47]: @jnsdls Can I stack them ser or do you want all of them in one PR?

Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/overview/components/ContractChecklist.tsx:47]: one pr ideally for this

**Review by jnsdls — APPROVED** [review]

**Pull request merged**